### ISSUE TITLE
Problem when inserting straight at top of list

### ISSUE THREAD

**Issue by neutralino1** [swanandp/acts_as_list#109]
Hi all,

Let's define
```
>>> p_1 = Item.find(1)
>>> p_2 = Item.find(2)
```
Say I do
```
>>> p_1.insert_at(1)
>>> p_2.insert_at(1)
```
Everything works as expected :
```
>>> p_1.reload.position
2
>>> p_2.position
1
```
`p_2` replaces `p_1` at the top of the list and `p_1` gets pushed down.

However, if I start again with `nil` positions and do
```
>>> p_1.position = 1
>>> p_1.save
>>> p_2.position = 1
>>> p_2.save
```
Then I end up with
```
>>> p_1.reload.position
0
>>> p_2.reload.position
1
```

I think the problems stems from the `update_positions` [method](https://github.com/swanandp/acts_as_list/blob/master/lib/acts_as_list/active_record/acts/list.rb#L420) that is called after update.

The method does
```
old_position = send("#{position_column}_was").to_i
```
And `nil.to_i` yields 0.

Thoughts?

**Comment by brendon** [comment]
@neutralino1, I'm trying to write a test to reproduce this problem but haven't been able to. Here's what I have so far. Can you fill in the gaps?

```ruby
class NilPositionTest < ActsAsListTestCase

  def setup
    setup_db
  end

  def test_nil_position_ordering
    new1 = DefaultScopedMixin.create pos: nil
    new2 = DefaultScopedMixin.create pos: nil

    new1.pos = 1
    new1.save

    new2.pos = 1
    new2.save

    assert_equal [1, 2], DefaultScopedMixin.all.map(&:pos)
  end

end
```

**Comment by brendon** [comment]
I have a test that reproduces the problem, except that I get `[1,1]` instead of `[0,1]`. Perhaps some changes in `master` since this issue was raised changed how the bug materialises?

**Comment by StoneFrog** [comment]
Hey guys, I managed to write test that reproduces the problem. I also did small change in code that seems to fix this issue - PR #230
